src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/Country.java

```diff
@@ -0,0 +1,279 @@
+    PAPUA_NEW_GUINEA("Papua New Guinea"),
+    PARAGUAY("Paraguay"),
+    PERU("Peru"),
+    PHILIPPINES("Philippines"),
+    PITCAIRN("Pitcairn"),
+    POLAND("Poland"),
+    PORTUGAL("Portugal"),
+    PUERTO_RICO("Puerto Rico"),
+    QATAR("Qatar"),
+    REUNION("Reunion"),
+    ROMANIA("Romania"),
+    RUSSIA("Russia"),
+    RWANDA("Rwanda"),
+    SAINT_BARTHELEMY("Saint Barthelemy"),
+    SAINT_HELENA_ASCENSION_AND_TRISTAN_DA_CUNHA("Saint Helena, Ascension and Tristan da Cunha"),
+    SAINT_KITTS_AND_NEVIS("Saint Kitts and Nevis"),
+    SAINT_LUCIA("Saint Lucia"),
+    SAINT_MARTIN_FRENCH_PART("Saint Martin (French part)"),
+    SAINT_PIERRE_AND_MIQUELON("Saint Pierre and Miquelon"),
+    SAINT_VINCENT_AND_THE_GRENADINES("Saint Vincent and the Grenadines"),
+    SAMOA("Samoa"),
+    SAN_MARINO("San Marino"),
+    SAO_TOME_AND_PRINCIPE("Sao Tome and Principe"),
+    SARK("Sark"),
+    SAUDI_ARABIA("Saudi Arabia"),
+    SENEGAL("Senegal"),
+    SERBIA("Serbia"),
+    SEYCHELLES("Seychelles"),
+    SIERRA_LEONE("Sierra Leone"),
+    SINGAPORE("Singapore"),
+    SINT_MAARTEN_DUTCH_PART("Sint Maarten (Dutch part)"),
+    SLOVAKIA("Slovakia"),
+    SLOVENIA("Slovenia"),
+    SOLOMON_ISLANDS("Solomon Islands"),
+    SOMALIA("Somalia"),
+    SOUTH_AFRICA("South Africa"),
+    SOUTH_GEORGIA_AND_THE_SOUTH_SANDWICH_ISLANDS("South Georgia and the South Sandwich Islands"),
+    SOUTH_KOREA("South Korea"),
+    SOUTH_SUDAN("South Sudan"),
+    SPAIN("Spain"),
+    SRI_LANKA("Sri Lanka"),
+    SUDAN("Sudan"),
+    SURINAME("Suriname"),
+    SVALBARD_AND_JAN_MAYEN("Svalbard and Jan Mayen"),
+    SWEDEN("Sweden"),
+    SWITZERLAND("Switzerland"),
+    SYRIA("Syria"),
+    TAIWAN_PROVINCE_OF_CHINA("Taiwan, Province of China"),
+    TAJIKISTAN("Tajikistan"),
+    TANZANIA("Tanzania"),
+    THAILAND("Thailand"),
+    TOGO("Togo"),
+    TOKELAU("Tokelau"),
+    TONGA("Tonga"),
+    TRINIDAD_AND_TOBAGO("Trinidad and Tobago"),
+    TUNISIA("Tunisia"),
+    TURKEY("Turkey"),
+    TURKMENISTAN("Turkmenistan"),
+    TURKS_AND_CAICOS_ISLANDS("Turks and Caicos Islands"),
+    TUVALU("Tuvalu"),
+    UGANDA("Uganda"),
+    UKRAINE("Ukraine"),
+    UNITED_ARAB_EMIRATES("United Arab Emirates"),
+    UNITED_STATES("United States"),
+    UNITED_STATES_MINOR_OUTLYING_ISLANDS("United States Minor Outlying Islands"),
+    URUGUAY("Uruguay"),
+    UZBEKISTAN("Uzbekistan"),
+    VANUATU("Vanuatu"),
+    VATICAN_CITY("Vatican City"),
+    VENEZUELA("Venezuela"),
+    VIETNAM("Vietnam"),
+    VIRGIN_ISLANDS_BRITISH("Virgin Islands, British"),
+    VIRGIN_ISLANDS_U_S("Virgin Islands, U.S."),
+    WALLIS_AND_FUTUNA("Wallis and Futuna"),
+    WESTERN_SAHARA("Western Sahara"),
+    YEMEN("Yemen"),
+    ZAMBIA("Zambia"),
+    ZIMBABWE("Zimbabwe");
+
+    public final String countryName;
+
+    Country(String country) {
+        this.countryName = country;
+    }
+
+    public static boolean isValid(String countryIn, String location, Errors errs, String loggingContext) {
```

Comment:
We need to add a validation with all countries, including "United Kingdom","England","Wales","Scotland" and "Northern Ireland", and not only oversees countries.

The spec here https://github.com/companieshouse/overseas-entities-api/blob/7510143c3d23510807846c90b2d687ba8411e17a/spec/schema.json#L1169 has to change.
```
"anyOf": [
    {
        "enum": [
        "United Kingdom",
        "England",
        "Scotland",
        "Wales",
        "Northern Ireland"
        ]
    },
    {
    ...
    }
]
```

Reply:
As agreed, this is not in the spec for the moment as it's not related to validation of the Overseas Entity information, so doesn't need to be implemented here yet either.

Reply:
Nope, it is related to the validation of overseas entity. We need to differentiate between `incorporation_country` field and `principal-address` field

Reply:
I will do it